modules/azure/storage.go

```diff
@@ -0,0 +1,246 @@
+		return "", err2
+	}
+	storageAccount, err := GetStorageAccountPropertyE(storageAccountName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return "", err
+	}
+	return string(storageAccount.Kind), nil
+}
+
+// GetStorageAccountSkuTierE returns the storage account sku tier as Standard or Premium.
+// This function would fail the test if there is an error.
+func GetStorageAccountSkuTierE(storageAccountName, resourceGroupName, subscriptionID string) (string, error) {
+	storageAccount, err := GetStorageAccountPropertyE(storageAccountName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return "", err
+	}
+	return string(storageAccount.Sku.Tier), nil
+}
+
+// GetStorageBlobContainerE returns Blob container client.
+// This function would fail the test if there is an error.
+func GetStorageBlobContainerE(containerName, storageAccountName, resourceGroupName, subscriptionID string) (*storage.BlobContainer, error) {
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	resourceGroupName, err2 := getTargetAzureResourceGroupName((resourceGroupName))
+	if err2 != nil {
+		return nil, err2
+	}
+	client, err := GetStorageBlobContainerClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	container, err := client.Get(context.Background(), resourceGroupName, storageAccountName, containerName)
+	if err != nil {
+		return nil, err
+	}
+	return &container, nil
+}
+
+// GetStorageAccountPropertyE returns StorageAccount properties.
+// This function would fail the test if there is an error.
+func GetStorageAccountPropertyE(storageAccountName, resourceGroupName, subscriptionID string) (*storage.Account, error) {
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	resourceGroupName, err2 := getTargetAzureResourceGroupName((resourceGroupName))
+	if err2 != nil {
+		return nil, err2
+	}
+	client, err := GetStorageAccountClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	account, err := client.GetProperties(context.Background(), resourceGroupName, storageAccountName, "")
+	if err != nil {
+		return nil, err
+	}
+	return &account, nil
+}
+
+// GetStorageAccountClientE creates a storage account client.
+// This function would fail the test if there is an error.
+func GetStorageAccountClientE(subscriptionID string) (*storage.AccountsClient, error) {
+	storageAccountClient := storage.NewAccountsClient(os.Getenv(AzureSubscriptionID))
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+	storageAccountClient.Authorizer = *authorizer
+	return &storageAccountClient, nil
+}
+
+// GetStorageBlobContainerClientE creates a storage container client.
+// This function would fail the test if there is an error.
+func GetStorageBlobContainerClientE(subscriptionID string) (*storage.BlobContainersClient, error) {
+	blobContainerClient := storage.NewBlobContainersClient(os.Getenv(AzureSubscriptionID))
```

Comment:
need to use getTargetAzureSubscription to get subscription id for consistency.

	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
	if err != nil {
		return nil, err
	}